library/std/src/panicking.rs

```diff
@@ -241,33 +243,54 @@ pub mod panic_count {
     // thread, if that thread currently views `GLOBAL_PANIC_COUNT` as being zero,
     // then `LOCAL_PANIC_COUNT` in that thread is zero. This invariant holds before
     // and after increase and decrease, but not necessarily during their execution.
+    //
+    // Addiitonally, the top bit of GLOBAL_PANIC_COUNT (GLOBAL_ALWAYS_ABORT_FLAG)
+    // records whether panic::always_abort() has been called.  This can only be
+    // set, never cleared.
+    //
+    // The same encoding is used for the return value from increase() and get().
+    // This could be viewed as a struct containing a single bit and an n-1-bit
+    // value, but if we wrote it like that it would be more than a single word.
+    // Even a newtype around usize would be clumsy because we need atomics.
+    //
+    // Stealing a bit is fine because it just amounts to assuming that each
+    // panicking thread consumes at least 2 bytes of address space.
     static GLOBAL_PANIC_COUNT: AtomicUsize = AtomicUsize::new(0);
 
+    pub fn is_always_panic(value: usize) -> bool {
+        value & ALWAYS_ABORT_FLAG != 0
+    }
+
     pub fn increase() -> usize {
```

Comment:
Maybe it makes things a bit easier to read if this returns `(usize, bool)`. Then the code below can do
```rust
let (panics, must_abort) = panic_count::increase();
```

Reply:
> Maybe it makes things a bit easier to read if this returns `(usize, bool)`. Then the code below can do
> 
> ```rust
> let (panics, must_abort) = panic_count::increase();
> ```

Mmm.  I tried that.  It is rather more typesafe.  I had a reason for changing it but now that I try to explain the reason it seems to make no sense.

I think changing this will fix a couple of your other comments too...